mlflow/java/spark/src/main/scala/org/mlflow/spark/autologging/DatasourceAttributeExtractor.scala

```diff
@@ -15,7 +21,7 @@ private[autologging] case class SparkTableInfo(
     formatOpt: Option[String])
 
 /** Helper object for extracting Spark datasource attributes from a Spark logical plan. */
-private[autologging] object DatasourceAttributeExtractor {
```

Comment:
The change here is turning existing datasource-attribute-extracting logic from a singleton object to a trait, with its behavior overridden in repl-ID-aware environments

Reply:
You might want to update the comment to reflect it's change from a singleton to a trait.